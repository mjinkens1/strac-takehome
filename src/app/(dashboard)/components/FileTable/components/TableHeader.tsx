export function TableHeader() {
  return (
    <thead className="sticky top-0 z-10 border-b border-[var(--color-card-border)] bg-[var(--color-card-bg)] text-left text-muted-foreground font-bold">
      <tr>
        <th className="px-6 py-4 w-[40%]">Filename</th>
        <th className="px-6 py-4 w-[25%]">Type</th>
        <th className="px-6 py-4 w-[25%]">Modified</th>
        <th className="px-6 py-4 w-[10%] text-right">Actions</th>
      </tr>
    </thead>
  );
}
