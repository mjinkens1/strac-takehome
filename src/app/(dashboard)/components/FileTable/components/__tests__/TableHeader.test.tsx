import { render, screen } from "@testing-library/react";
import { TableHeader } from "../TableHeader";

describe("TableHeader", () => {
  it("renders all column headers", () => {
    render(<TableHeader />);

    const headers = screen.getAllByRole("columnheader");
    expect(headers).toHaveLength(4);

    expect(headers[0]).toHaveTextContent("Filename");
    expect(headers[1]).toHaveTextContent("Type");
    expect(headers[2]).toHaveTextContent("Modified");
    expect(headers[3]).toHaveTextContent("Actions");
  });

  it("has correct styling classes", () => {
    render(<TableHeader />);

    const thead = screen.getByRole("rowgroup");
    expect(thead).toHaveClass(
      "sticky",
      "top-0",
      "z-10",
      "border-b",
      "border-[var(--color-card-border)]",
      "bg-[var(--color-card-bg)]"
    );
  });

  it("has correct width classes for columns", () => {
    render(<TableHeader />);

    const headers = screen.getAllByRole("columnheader");
    expect(headers[0]).toHaveClass("w-[40%]");
    expect(headers[1]).toHaveClass("w-[25%]");
    expect(headers[2]).toHaveClass("w-[25%]");
    expect(headers[3]).toHaveClass("w-[10%]");
  });

  it("matches snapshot", () => {
    const { container } = render(<TableHeader />);
    expect(container).toMatchSnapshot();
  });
});
