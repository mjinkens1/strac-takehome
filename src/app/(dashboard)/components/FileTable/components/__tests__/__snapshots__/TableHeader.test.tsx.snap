// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableHeader matches snapshot 1`] = `
<div>
  <thead
    class="sticky top-0 z-10 border-b border-[var(--color-card-border)] bg-[var(--color-card-bg)] text-left text-muted-foreground font-bold"
  >
    <tr>
      <th
        class="px-6 py-4 w-[40%]"
      >
        Filename
      </th>
      <th
        class="px-6 py-4 w-[25%]"
      >
        Type
      </th>
      <th
        class="px-6 py-4 w-[25%]"
      >
        Modified
      </th>
      <th
        class="px-6 py-4 w-[10%] text-right"
      >
        Actions
      </th>
    </tr>
  </thead>
</div>
`;
